src/db_adapters/receipts.rs

```diff
@@ -13,16 +14,38 @@ use crate::schema;
 
 /// Saves receipts to database
 pub(crate) async fn store_receipts(
+    pool: &actix_diesel::Database<PgConnection>,
+    shards: &[near_indexer::IndexerShard],
+    block_hash: &near_indexer::near_primitives::hash::CryptoHash,
+    block_timestamp: u64,
+    strict_mode: bool,
+) -> anyhow::Result<()> {
+    let futures = shards
+        .iter()
+        .filter_map(|shard| shard.chunk.as_ref())
+        .filter(|chunk| !chunk.receipts.is_empty())
+        .map(|chunk| {
+            store_chunk_receipts(
+                pool,
+                &chunk.receipts,
+                block_hash,
+                &chunk.header.chunk_hash,
+                block_timestamp,
+                strict_mode,
+            )
+        });
+
+    try_join_all(futures).await.map(|_| ())
+}
+
+async fn store_chunk_receipts(
     pool: &actix_diesel::Database<PgConnection>,
     receipts: &[near_indexer::near_primitives::views::ReceiptView],
-    block_hash: &str,
+    block_hash: &near_indexer::near_primitives::hash::CryptoHash,
     chunk_hash: &near_indexer::near_primitives::hash::CryptoHash,
     block_timestamp: u64,
     strict_mode: bool,
 ) -> anyhow::Result<()> {
-    if receipts.is_empty() {
```

Comment:
Why did you remove this?

Reply:
I moved this check into line 26 (`filter`)